### ISSUE TITLE
Cast_control no longer working with my Chromecast Audio

### ISSUE THREAD

**Issue by b0o** [alexdelorenzo/cast_control#12]
When this package was still called `chromecast_mpris` it worked with perfectly with my Chromecast Audio. It's no longer working and I'm now seeing the following error messages:

```
Sep 06 17:29:31 systemd[1606]: Stopping Provides Chromecast player control via MPRIS...
Sep 06 17:29:31 systemd[1606]: Stopped Provides Chromecast player control via MPRIS.
Sep 06 17:29:31 systemd[1606]: Started Provides Chromecast player control via MPRIS.
Sep 06 17:29:36 cast_control[700592]: DEBUG:root:Connecting to D-Bus session bus...
Sep 06 17:29:36 cast_control[700592]: DEBUG:pychromecast.socket_client:[Chromecast Audio(xxx.xxx.xxx.xxx):8009] Received: Message urn:x-cast:com.google.cast.media from xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx to sender-0: {'type': 'MEDIA_STATUS', 'status': [{'mediaSessionId': 1, 'playbackRate': 1, 'playerState': 'PLAYING', 'currentTime': 7.628, 'supportedMediaCommands': 514511, 'volume': {'level': 1, 'muted': False}, 'activeTrackIds': [], 'media': {'contentId': 'spotify:track:5xXqi5Yc55K8NQocuMpBMt', 'streamType': 'BUFFERED', 'mediaCategory': 'AUDIO', 'contentType': 'application/x-spotify.track', 'metadata': {'metadataType': 3, 'title': 'Slackers', 'songName': 'Slackers', 'artist': 'Daniel Koestner', 'albumName': 'Donut County (Original Soundtrack)', 'images': [{'url': 'https://i.scdn.co/image/ab67616d00001e024b60a62bcf083f014eb86b32', 'height': 300, 'width': 300}, {'url': 'https://i.scdn.co/image/ab67616d000048514b60a62bcf083f014eb86b32', 'height': 64, 'width': 64}, {'url': 'https://i.scdn.co/image/ab67616d0000b2734b60a62bcf083f014eb86b32', 'height': 640, 'width': 640}]}, 'entity': 'spotify:track:5xXqi5Yc55K8NQocuMpBMt', 'duration': 99.31, 'breaks': [], 'breakClips': []}, 'currentItemId': 1, 'items': [{'itemId': 1, 'media': {'contentId': '', 'streamType': 'NONE', 'contentType': '', 'duration': 274.48}, 'autoplay': False, 'orderId': 0}], 'repeatMode': 'REPEAT_OFF'}], 'requestId': 3}
Sep 06 17:29:36 cast_control[700592]: DEBUG:pychromecast.controllers:Media:Received status {'type': 'MEDIA_STATUS', 'status': [{'mediaSessionId': 1, 'playbackRate': 1, 'playerState': 'PLAYING', 'currentTime': 7.628, 'supportedMediaCommands': 514511, 'volume': {'level': 1, 'muted': False}, 'activeTrackIds': [], 'media': {'contentId': 'spotify:track:5xXqi5Yc55K8NQocuMpBMt', 'streamType': 'BUFFERED', 'mediaCategory': 'AUDIO', 'contentType': 'application/x-spotify.track', 'metadata': {'metadataType': 3, 'title': 'Slackers', 'songName': 'Slackers', 'artist': 'Daniel Koestner', 'albumName': 'Donut County (Original Soundtrack)', 'images': [{'url': 'https://i.scdn.co/image/ab67616d00001e024b60a62bcf083f014eb86b32', 'height': 300, 'width': 300}, {'url': 'https://i.scdn.co/image/ab67616d000048514b60a62bcf083f014eb86b32', 'height': 64, 'width': 64}, {'url': 'https://i.scdn.co/image/ab67616d0000b2734b60a62bcf083f014eb86b32', 'height': 640, 'width': 640}]}, 'entity': 'spotify:track:5xXqi5Yc55K8NQocuMpBMt', 'duration': 99.31, 'breaks': [], 'breakClips': []}, 'currentItemId': 1, 'items': [{'itemId': 1, 'media': {'contentId': '', 'streamType': 'NONE', 'contentType': '', 'duration': 274.48}, 'autoplay': False, 'orderId': 0}], 'repeatMode': 'REPEAT_OFF'}], 'requestId': 3}
Sep 06 17:29:36 cast_control[700592]: INFO:root:MPRIS server connected to D-Bus session bus
Sep 06 17:29:37 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.PlaybackStatus() called.
Sep 06 17:29:37 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.LoopStatus() called.
Sep 06 17:29:37 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.Rate() called.
Sep 06 17:29:37 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.Shuffle() called.
Sep 06 17:29:37 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.Metadata() called.
Sep 06 17:29:37 cast_control[700592]: /usr/lib/python3.9/site-packages/gi/overrides/GLib.py:118: Warning: g_variant_new_object_path: assertion 'g_variant_is_object_path (object_path)' failed
Sep 06 17:29:37 cast_control[700592]:   return self._LEAF_CONSTRUCTORS[format](value)
Sep 06 17:29:37 cast_control[700592]: ERROR:pydbus.registration:Exception while handling org.freedesktop.DBus.Properties.GetAll()
Sep 06 17:29:37 cast_control[700592]: Traceback (most recent call last):
Sep 06 17:29:37 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/pydbus/registration.py", line 81, in call_method
Sep 06 17:29:37 cast_control[700592]:     result = method(*parameters, **kwargs)
Sep 06 17:29:37 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/pydbus/registration.py", line 110, in GetAll
Sep 06 17:29:37 cast_control[700592]:     ret[local] = GLib.Variant(type, getattr(self.object, local))
Sep 06 17:29:37 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/interfaces/interface.py", line 26, in new_method
Sep 06 17:29:37 cast_control[700592]:     return method(self, *args, **kwargs)
Sep 06 17:29:37 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 280, in Metadata
Sep 06 17:29:37 cast_control[700592]:     metadata: DbusMetadata = self._dbus_metadata()
Sep 06 17:29:37 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 74, in _dbus_metadata
Sep 06 17:29:37 cast_control[700592]:     return get_dbus_metadata(metadata)
Sep 06 17:29:37 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 165, in get_dbus_metadata
Sep 06 17:29:37 cast_control[700592]:     return {
Sep 06 17:29:37 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 166, in <dictcomp>
Sep 06 17:29:37 cast_control[700592]:     entry: get_dbus_var(entry, val)
Sep 06 17:29:37 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 156, in get_dbus_var
Sep 06 17:29:37 cast_control[700592]:     return Variant(METADATA_TYPES[entry], obj)
Sep 06 17:29:37 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/gi/overrides/GLib.py", line 189, in __new__
Sep 06 17:29:37 cast_control[700592]:     v = creator._create(format_string, value)
Sep 06 17:29:37 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/gi/overrides/GLib.py", line 118, in _create
Sep 06 17:29:37 cast_control[700592]:     return self._LEAF_CONSTRUCTORS[format](value)
Sep 06 17:29:37 cast_control[700592]: TypeError: constructor returned NULL
Sep 06 17:29:38 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.PlaybackStatus() called.
Sep 06 17:29:38 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.LoopStatus() called.
Sep 06 17:29:38 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.Rate() called.
Sep 06 17:29:38 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.Shuffle() called.
Sep 06 17:29:38 cast_control[700592]: DEBUG:root:org.mpris.MediaPlayer2.Player.Metadata() called.
Sep 06 17:29:38 cast_control[700592]: ERROR:pydbus.registration:Exception while handling org.freedesktop.DBus.Properties.GetAll()
Sep 06 17:29:38 cast_control[700592]: Traceback (most recent call last):
Sep 06 17:29:38 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/pydbus/registration.py", line 81, in call_method
Sep 06 17:29:38 cast_control[700592]:     result = method(*parameters, **kwargs)
Sep 06 17:29:38 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/pydbus/registration.py", line 110, in GetAll
Sep 06 17:29:38 cast_control[700592]:     ret[local] = GLib.Variant(type, getattr(self.object, local))
Sep 06 17:29:38 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/interfaces/interface.py", line 26, in new_method
Sep 06 17:29:38 cast_control[700592]:     return method(self, *args, **kwargs)
Sep 06 17:29:38 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 280, in Metadata
Sep 06 17:29:38 cast_control[700592]:     metadata: DbusMetadata = self._dbus_metadata()
Sep 06 17:29:38 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 74, in _dbus_metadata
Sep 06 17:29:38 cast_control[700592]:     return get_dbus_metadata(metadata)
Sep 06 17:29:38 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 165, in get_dbus_metadata
Sep 06 17:29:38 cast_control[700592]:     return {
Sep 06 17:29:38 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 166, in <dictcomp>
Sep 06 17:29:38 cast_control[700592]:     entry: get_dbus_var(entry, val)
Sep 06 17:29:38 cast_control[700592]:   File "/home/user/.local/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 156, in get_dbus_var
Sep 06 17:29:38 cast_control[700592]:     return Variant(METADATA_TYPES[entry], obj)
Sep 06 17:29:38 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/gi/overrides/GLib.py", line 189, in __new__
Sep 06 17:29:38 cast_control[700592]:     v = creator._create(format_string, value)
Sep 06 17:29:38 cast_control[700592]:   File "/usr/lib/python3.9/site-packages/gi/overrides/GLib.py", line 118, in _create
Sep 06 17:29:38 cast_control[700592]:     return self._LEAF_CONSTRUCTORS[format](value)
Sep 06 17:29:38 cast_control[700592]: TypeError: constructor returned NULL
```

I'm on Arch Linux and using the cast_control AUR package.

**Comment by b0o** [comment]
I just tried the version from pip, still seeing the errors:

```
Sep 06 17:40:39 systemd[1606]: Started Provides Chromecast player control via MPRIS.
Sep 06 17:40:45 env[909322]: DEBUG:root:Connecting to D-Bus session bus...
Sep 06 17:40:45 env[909322]: INFO:root:MPRIS server connected to D-Bus session bus
Sep 06 17:40:45 env[909322]: DEBUG:pychromecast.socket_client:[(xxx.xxx.xxx.xxx):8009] Received: Message urn:x-cast:com.google.youtube.mdx from xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx to sender-0: {'type': 'mdxSessionStatus', 'data': {'screenId': 'tf75c1l2bqgnatgs5vtfht69c', 'deviceId': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', 'loungeToken': 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx', 'loungeTokenRefreshIntervalMs': 1500000}}
Sep 06 17:40:45 env[909322]: DEBUG:pychromecast.socket_client:[(xxx.xxx.xxx.xxx):8009] Received: Message urn:x-cast:com.google.cast.media from xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx to *: {'type': 'MEDIA_STATUS', 'status': [{'mediaSessionId': 1152415589, 'playbackRate': 1, 'supportedMediaCommands': 262275, 'volume': {'level': 0.6100000143051147, 'muted': False}, 'playerState': 'PLAYING', 'customData': {'playerState': 1}, 'currentTime': 9096.276, 'media': {'contentId': 'pSvtQZ2uHuw', 'contentType': 'x-youtube/video', 'customData': {'listId': 'RQO0kHYC53vBguhUU9BPuSSs7hpiQ', 'currentIndex': 4}, 'streamType': 'BUFFERED', 'metadata': {'metadataType': 0, 'title': 'City Park Sounds for Study / Studying in the Park #270', 'subtitle': 'Winter Whale', 'images': [{'url': 'https://i.ytimg.com/vi/pSvtQZ2uHuw/hqdefault.jpg'}]}, 'duration': 11260.261587301587}}]}
Sep 06 17:40:45 env[909322]: DEBUG:pychromecast.controllers:Media:Received status {'type': 'MEDIA_STATUS', 'status': [{'mediaSessionId': 1152415589, 'playbackRate': 1, 'supportedMediaCommands': 262275, 'volume': {'level': 0.6100000143051147, 'muted': False}, 'playerState': 'PLAYING', 'customData': {'playerState': 1}, 'currentTime': 9096.276, 'media': {'contentId': 'pSvtQZ2uHuw', 'contentType': 'x-youtube/video', 'customData': {'listId': 'RQO0kHYC53vBguhUU9BPuSSs7hpiQ', 'currentIndex': 4}, 'streamType': 'BUFFERED', 'metadata': {'metadataType': 0, 'title': 'City Park Sounds for Study / Studying in the Park #270', 'subtitle': 'Winter Whale', 'images': [{'url': 'https://i.ytimg.com/vi/pSvtQZ2uHuw/hqdefault.jpg'}]}, 'duration': 11260.261587301587}}]}
Sep 06 17:40:45 env[909322]: DEBUG:root:org.mpris.MediaPlayer2.Player.Volume() called.
Sep 06 17:40:45 env[909322]: DEBUG:root:org.mpris.MediaPlayer2.Player.Metadata() called.
Sep 06 17:40:45 env[909322]: /home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/gi/overrides/GLib.py:118: Warning: g_variant_new_object_path: assertion 'g_variant_is_object_path (object_path)' failed
Sep 06 17:40:45 env[909322]:   return self._LEAF_CONSTRUCTORS[format](value)
Sep 06 17:40:45 env[909322]: ERROR:pychromecast.controllers.media:Exception thrown when calling media status callback
Sep 06 17:40:45 env[909322]: Traceback (most recent call last):
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/pychromecast/controllers/media.py", line 488, in _fire_status_changed
Sep 06 17:40:45 env[909322]:     listener.new_media_status(self.status)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/cast_control/device/listeners.py", line 82, in new_media_status
Sep 06 17:40:45 env[909322]:     self._update_metadata(status)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/cast_control/device/listeners.py", line 71, in _update_metadata
Sep 06 17:40:45 env[909322]:     self._update_volume(status)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/cast_control/device/listeners.py", line 68, in _update_volume
Sep 06 17:40:45 env[909322]:     self.on_volume()
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/events.py", line 63, in on_volume
Sep 06 17:40:45 env[909322]:     self.emit_player_changes(ON_VOLUME_PROPS)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/events.py", line 57, in emit_player_changes
Sep 06 17:40:45 env[909322]:     self.emit_changes(self.player, changes)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/events.py", line 32, in emit_changes
Sep 06 17:40:45 env[909322]:     dbus_emit_changes(interface, changes)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/base.py", line 161, in dbus_emit_changes
Sep 06 17:40:45 env[909322]:     attr_vals: AttrVals = {
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/base.py", line 162, in <dictcomp>
Sep 06 17:40:45 env[909322]:     attr: getattr(interface, attr)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/interfaces/interface.py", line 26, in new_method
Sep 06 17:40:45 env[909322]:     return method(self, *args, **kwargs)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 280, in Metadata
Sep 06 17:40:45 env[909322]:     metadata: DbusMetadata = self._dbus_metadata()
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 74, in _dbus_metadata
Sep 06 17:40:45 env[909322]:     return get_dbus_metadata(metadata)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 165, in get_dbus_metadata
Sep 06 17:40:45 env[909322]:     return {
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 166, in <dictcomp>
Sep 06 17:40:45 env[909322]:     entry: get_dbus_var(entry, val)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 156, in get_dbus_var
Sep 06 17:40:45 env[909322]:     return Variant(METADATA_TYPES[entry], obj)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/gi/overrides/GLib.py", line 189, in __new__
Sep 06 17:40:45 env[909322]:     v = creator._create(format_string, value)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/gi/overrides/GLib.py", line 118, in _create
Sep 06 17:40:45 env[909322]:     return self._LEAF_CONSTRUCTORS[format](value)
Sep 06 17:40:45 env[909322]: TypeError: constructor returned NULL
Sep 06 17:40:45 env[909322]: DEBUG:pychromecast.socket_client:[(xxx.xxx.xxx.xxx):8009] Received: Message urn:x-cast:com.google.cast.media from xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx to sender-0: {'type': 'MEDIA_STATUS', 'requestId': 3, 'status': [{'mediaSessionId': 1152415589, 'playbackRate': 1, 'supportedMediaCommands': 262275, 'volume': {'level': 0.6100000143051147, 'muted': False}, 'playerState': 'PLAYING', 'customData': {'playerState': 1}, 'currentTime': 9096.292, 'media': {'contentId': 'pSvtQZ2uHuw', 'contentType': 'x-youtube/video', 'customData': {'listId': 'RQO0kHYC53vBguhUU9BPuSSs7hpiQ', 'currentIndex': 4}, 'streamType': 'BUFFERED', 'metadata': {'metadataType': 0, 'title': 'City Park Sounds for Study / Studying in the Park #270', 'subtitle': 'Winter Whale', 'images': [{'url': 'https://i.ytimg.com/vi/pSvtQZ2uHuw/hqdefault.jpg'}]}, 'duration': 11260.261587301587}}]}
Sep 06 17:40:45 env[909322]: DEBUG:pychromecast.controllers:Media:Received status {'type': 'MEDIA_STATUS', 'requestId': 3, 'status': [{'mediaSessionId': 1152415589, 'playbackRate': 1, 'supportedMediaCommands': 262275, 'volume': {'level': 0.6100000143051147, 'muted': False}, 'playerState': 'PLAYING', 'customData': {'playerState': 1}, 'currentTime': 9096.292, 'media': {'contentId': 'pSvtQZ2uHuw', 'contentType': 'x-youtube/video', 'customData': {'listId': 'RQO0kHYC53vBguhUU9BPuSSs7hpiQ', 'currentIndex': 4}, 'streamType': 'BUFFERED', 'metadata': {'metadataType': 0, 'title': 'City Park Sounds for Study / Studying in the Park #270', 'subtitle': 'Winter Whale', 'images': [{'url': 'https://i.ytimg.com/vi/pSvtQZ2uHuw/hqdefault.jpg'}]}, 'duration': 11260.261587301587}}]}
Sep 06 17:40:45 env[909322]: DEBUG:root:org.mpris.MediaPlayer2.Player.Volume() called.
Sep 06 17:40:45 env[909322]: DEBUG:root:org.mpris.MediaPlayer2.Player.Metadata() called.
Sep 06 17:40:45 env[909322]: ERROR:pychromecast.controllers.media:Exception thrown when calling media status callback
Sep 06 17:40:45 env[909322]: Traceback (most recent call last):
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/pychromecast/controllers/media.py", line 488, in _fire_status_changed
Sep 06 17:40:45 env[909322]:     listener.new_media_status(self.status)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/cast_control/device/listeners.py", line 82, in new_media_status
Sep 06 17:40:45 env[909322]:     self._update_metadata(status)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/cast_control/device/listeners.py", line 71, in _update_metadata
Sep 06 17:40:45 env[909322]:     self._update_volume(status)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/cast_control/device/listeners.py", line 68, in _update_volume
Sep 06 17:40:45 env[909322]:     self.on_volume()
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/events.py", line 63, in on_volume
Sep 06 17:40:45 env[909322]:     self.emit_player_changes(ON_VOLUME_PROPS)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/events.py", line 57, in emit_player_changes
Sep 06 17:40:45 env[909322]:     self.emit_changes(self.player, changes)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/events.py", line 32, in emit_changes
Sep 06 17:40:45 env[909322]:     dbus_emit_changes(interface, changes)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/base.py", line 161, in dbus_emit_changes
Sep 06 17:40:45 env[909322]:     attr_vals: AttrVals = {
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/base.py", line 162, in <dictcomp>
Sep 06 17:40:45 env[909322]:     attr: getattr(interface, attr)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/interfaces/interface.py", line 26, in new_method
Sep 06 17:40:45 env[909322]:     return method(self, *args, **kwargs)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 280, in Metadata
Sep 06 17:40:45 env[909322]:     metadata: DbusMetadata = self._dbus_metadata()
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/interfaces/player.py", line 74, in _dbus_metadata
Sep 06 17:40:45 env[909322]:     return get_dbus_metadata(metadata)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 165, in get_dbus_metadata
Sep 06 17:40:45 env[909322]:     return {
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 166, in <dictcomp>
Sep 06 17:40:45 env[909322]:     entry: get_dbus_var(entry, val)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/mpris_server/mpris/metadata.py", line 156, in get_dbus_var
Sep 06 17:40:45 env[909322]:     return Variant(METADATA_TYPES[entry], obj)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/gi/overrides/GLib.py", line 189, in __new__
Sep 06 17:40:45 env[909322]:     v = creator._create(format_string, value)
Sep 06 17:40:45 env[909322]:   File "/home/user/.local/pipx/venvs/cast-control/lib/python3.9/site-packages/gi/overrides/GLib.py", line 118, in _create
Sep 06 17:40:45 env[909322]:     return self._LEAF_CONSTRUCTORS[format](value)
Sep 06 17:40:45 env[909322]: TypeError: constructor returned NULL
```

**Comment by alexdelorenzo** [comment]
Thanks for opening an issue, @b0o. Sorry, looks like I pushed a bad update to PyPI. Try downgrading `mpris_server` and `cast_control` and reinstall the last stable versions like so:

```bash
python3 -m pip uninstall mpris_server cast_control
python3 -m pip uninstall app_paths unpackable  # these are deps for cast_control

python3 -m pip install mpris_server==0.4.2 cast_control==0.11.4
```

I don't use `pipx` so I'm not sure if that will work correctly with your setup.

I've yanked every release after `cast_control==0.11.4` and `mpris_server==0.4.2` so other users don't run into this problem, so thanks again for bringing this to my attention.
